server/tserver/src/main/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystem.java

```diff
@@ -416,6 +444,12 @@ protected Status replicateLogs(ClientContext peerContext, final HostAndPort peer
     Status lastStatus = status, currentStatus = status;
     final AtomicReference<Exception> exceptionRef = new AtomicReference<>();
     while (true) {
+      // Check if the thread has been interrupted prior to replicating data
+      if (Thread.interrupted()) {
```

Comment:
Fair point; that is hard to say.  I'll add it regardless to make sure the thread will stop appropriately.